docs/SECURITY.md

```diff
@@ -0,0 +1,306 @@
+this bias is not relevant in the setting of using the DKG to perform digital
+signatures, which offers other strong properties -
+[paper](https://pdfs.semanticscholar.org/642b/d1bbc86c7750cef9fa770e9e4ba86bd49eb9.pdf).
+In particular, the paper mentions discrete log based systems. However, it is not
+yet strictly proven that system using computation co-CDH assumptions as required
+for threshold BLS signatures are secure in the model of Gennaro.  However, it is
+believed that this assumptions holds in this context as well and is being worked
+on.  Note that using threshold BLS signatures as a source of randomness _is_
+formally proven secured in this [paper](https://eprint.iacr.org/2020/096.pdf)
+from Galindo et al. 
+
+### Randomness generation model
+
+**Network**: The randomness generation protocol do not make any assumption on
+the network bounds. As soon as packet comes in, node processes them and the
+chain advances if conditions are there (enough partial beacon and time for a new round).
+
+**Synchronized Clocks**: All nodes must have roughly synchronized clocks to
+start the rounds at the same time. The precision of the synchronicity between
+clocks only needs to be at the order the round frequency (order of tens of
+seconds), which is much higher than the reality of server's clock (NTP-synced
+servers have drifts of under a second over the globe). XXX need source.
+
+**Broadcast channel**: The randomness generation models only needs a regular
+broadcast channel. It does **not** need to be reliable given the deterministic
+nature of a partial beacon and the drand chain: for a given round, a given node
+can only send one valid partial beacon.
+
+**Threshold**: The threshold is the amount of nodes that must be online and
+honest at a given time to broadcast their partial signature in order to create
+the final random beacon. See the attack section to know what are the
+consequences when that is not the case.
+
+**Determinism of the chain**: The chain is deterministic with respect to a fresh
+DKG phase. It means, if a trust party gathered a threshold of private shares, it
+could generate any beacon of the chain. When a resharing occurs, the individual
+shares of each drand node change but the chain remains the same as well. If the
+same set of nodes perform a new fresh DKG, it will create a new chain from
+scratch.
+
+**Resharing**: During a resharing, a drand network A (with threshold
+`tA`)creates new shares for a drand network B (with threshold `tB`) which can be
+disjoinct from A, such the drand network B is now responsible to continue
+creating drand beacons, while that the distributed public key doesn't change.
+For this to happen, there needs to be at least `tA` nodes from network A and
+`tB` nodes alive and honest during the resharing. At the end of the protocol,
+there is gonna be at least `tB` nodes that are qualified and have private shares
+to generate randomness.
+
+## Attacks
```

Comment:
```suggestion
## Threat Model
```

Reply:
This section is describing concrete attacs, this is different than describing the model under which these attacks occur (like network etc).

Reply:
```suggestion
## Attack Vectors
```